uber/menu.py

```diff
@@ -0,0 +1,51 @@
+from uber.common import *
+
+
+class MenuItem:
+    access = None   # list of permission levels allowed to display this menu
+    href = None     # link to render
+    submenu = None  # submenu to show
+    name = None     # name of Menu item to show
+
+    def __init__(self, href=None, access=None, submenu=None, name=None):
+        if submenu:
+            self.submenu = listify(submenu)
+        else:
+            self.href = href
+
+        self.name = name
+        self.access = access
+
+    def __getitem__(self, key):
+        for sm in self.submenu:
+            if sm.name == key:
+                return sm
+
+    def append_menu_item(self, m):
+        # if we aren't a submenu, convert us to one now
```

Comment:
This comment actually confused the heck outta me (I thought it meant that the function ONLY did submenus) and I had to go digging in your other PRs to see how this worked. Since this is something plugin authors will use, we need a really robust comment on this function.

Maybe something like:
```
###
Appends a new top-level menu item when called on a menu (e.g., c.MENU). If called on an existing menu item (e.g., c.MENU['People']), we convert that item to a submenu and/or append the menu item to the bottom of the list.
###